test/crud_test.rb

```diff
@@ -101,6 +106,45 @@ class ModelUpdateOperation < CreateOperation
     ModelUpdateOperation[{id: 1, song: {title: "Mercy Day For Mr. Vengeance"}}].model.must_equal song
   end
 
+  # model Song, :action
+  class FetchCollectionOperation < CreateOperation
+    model Song
+
+    contract do
+      property :title
+    end
+
+    def setup_model!(params)
+      if @user_role == 'admin'
```

Comment:
Sidenote: I don't think we need this differentiation between different roles here. It's good to see how you solve that problem, but a bit "off-topic" for this test, as we really only test fetching a collection here.